src/Shared/PackageManagers/NuGetProjectManager.cs

```diff
@@ -208,77 +140,26 @@ public override async Task<bool> PackageExists(PackageURL purl, bool useCache =
                 Logger.Trace("Provided PackageURL was null.");
                 return false;
             }
-            string packageName = purl.Name;
-            HttpClient httpClient = CreateHttpClient();
 
-            return await CheckJsonCacheForPackage(httpClient, $"{RegistrationEndpoint}{packageName.ToLowerInvariant()}/index.json", useCache);
+            return await Provider.DoesPackageExistAsync(purl, useCache: useCache);
         }
 
         /// <inheritdoc />
         public override async Task<IEnumerable<string>> EnumerateVersions(PackageURL purl, bool useCache = true)
         {
-            Logger.Trace("EnumerateVersions {0}", purl?.ToString());
+            Logger.Trace("EnumerateVersions {0}", purl.ToString());
 
-            if (purl == null || purl.Name is null)
+            if (purl.Name is null)
```

Comment:
Shouldn't this be `purl?.Name is null`?

Reply:
purl shouldn't ever be null. I did just change it though. Now it is `if (string.IsNullOrWhiteSpace(purl.Name))`